src/types/query.d.ts

```diff
@@ -173,29 +315,66 @@ export module query {
   export function Year(expr: ExprArg): Expr
   export function Month(expr: ExprArg): Expr
 
+  // TODO
   export function NextId(): Expr
   export function NewId(): Expr
   export function Database(name: ExprArg, scope?: ExprArg): Expr
-  export function Index(name: ExprArg, scope?: ExprArg): Expr
-  export function Class(name: ExprArg, scope?: ExprArg): Expr
-  export function Collection(name: ExprArg, scope?: ExprArg): Expr
-  export function Function(name: ExprArg, scope?: ExprArg): Expr
+
+  // TODO: "scope" argument
+  export function Index<T extends object>(
+    name: ExprVal<string>,
+    scope?: ExprArg
```

Comment:
`scope` is a `DatabaseRef`

Reply:
i'm not using fql with database refs currently, so this will remain untyped until someone contributes